app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+
+  validates :email, presence: true
+  validates :encrypted_password, presence: true
+  validates :username, presence: true
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :address, presence: true
+  validates :phone_number, presence: true
+end
```

Comment:
This is more of a comment about planning and Pivotal Tracker than it is about the code, but think about your story: "As a visitor, In order to place an order, I would like to create an account" -- Where does it say that you should have address, phone number and full name? If the purpose is just to create an account, you could just sign up with email and password, so these aren't really in the scope of the story as it is written.

I think the user wants to create an account with address and contact info so that the order comes to the right place and so that they can be contacted if something goes wrong with their order. Then the feature makes sense, and it would be clear to the developers, the product owner, the client, etc. more or less what you're doing and why. I realize that you were given these stories, but it's your project now, so it's up to you to update and edit them.